drivers/dma/dma_intel_adsp_hda.c

```diff
@@ -326,6 +326,18 @@ int intel_adsp_hda_dma_stop(const struct device *dev, uint32_t channel)
 
 	intel_adsp_hda_disable(cfg->base, cfg->regblock_size, channel);
 
+	/* host dma needs some cycles to completely stop */
+	if (cfg->direction == HOST_TO_MEMORY || cfg->direction == MEMORY_TO_HOST) {
+		uint32_t delay_cnt = 500;
+
+		do {
```

Comment:
the WAIT_FOR macro was made/copied from SOF for this exact sort of thing, https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/util.h#L760

Reply:
thanks, updated